text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,751 @@
+    modify the baseline expectations of a target, or split an existing target
+    into multiple targets with different baseline expectations. A proposal to
+    do so will be treated similarly to the analogous promotion, demotion, or
+    removal of a target, according to this policy, with the same team approvals
+    required.
+    - For instance, if an OS version has become obsolete and unsupported, a
+      target for that OS may raise its baseline expectations for OS version
+      (treated as though removing a target corresponding to the older
+      versions), or a target for that OS may split out support for older OS
+      versions into a lower-tier target (treated as though demoting a target
+      corresponding to the older versions, and requiring justification for a
+      new target at a lower tier for the older OS versions).
+- Tier 2 targets must not leave any significant portions of `core` or the
+  standard library unimplemented or stubbed out, unless they cannot possibly be
+  supported on the target.
+  - The right approach to handling a missing feature from a target may depend
+    on whether the target seems likely to develop the feature in the future. In
+    some cases, a target may be co-developed along with Rust support, and Rust
+    may gain new features on the target as that target gains the capabilities
+    to support those features.
+- The code generation backend for the target should not have deficiencies that
+  invalidate Rust safety properties, as evaluated by the Rust compiler team.
+  (This requirement does not apply to arbitrary security enhancements or
+  mitigations provided by code generation backends, only to those properties
+  needed to ensure safe Rust code cannot cause undefined behavior or other
+  unsoundness.) If this requirement does not hold, the target must clearly and
+  prominently document any such limitations as part of the target's entry in
+  the target tier list, and ideally also via a failing test in the testsuite.
+  The Rust compiler team must be satisfied with the balance between these
+  limitations and the difficulty of implementing the necessary features.
+  - For example, if Rust relies on a specific code generation feature to ensure
+    that safe code cannot overflow the stack, the code generation for the
+    target should support that feature.
+  - If the Rust compiler introduces new safety properties (such as via new
+    capabilities of a compiler backend), the Rust compiler team will determine
+    if they consider those new safety properties a best-effort improvement for
+    specific targets, or a required property for all Rust targets. In the
+    latter case, the compiler team may require the maintainers of existing
+    targets to either implement and confirm support for the property or update
+    the target tier list with documentation of the missing property.
+- If the target supports C code, the Rust target should support the C calling
```

Comment:
What is the reason this is a "should" and not a "must"? Notably I didn't see this becoming a "must" in the higher tiers, but that sounds like a requirement we want to have at _some_ point.

Reply:
I was concerned about the potential existence of platforms for which there's rudimentary C support but no interoperability, such as those supporting a limited C dialect, or those that only support compiling entire programs at once. I've upgraded this to a "must", and made it conditional on the existence of an interoperable calling convention for us to support.